scrapscript.py

```diff
@@ -1210,6 +1214,10 @@ def eval_exp(env: Env, exp: Object) -> Object:
             # captured environment with a binding to themselves.
             assert isinstance(value.env, dict)
             value.env[exp.name.name] = value
+            # If creating a Closure with an Assign we can specify a name for the
+            # Closure, which may be useful for debugging.
+            print("Setting closure with name ", exp.name.name)
+            value = Closure(value.env, value.func, name=exp.name.name)
```

Comment:
If you do

```
a = x -> x
b = a
```

what happens?

Reply:
An alternative would be catching this in the parser where we see a pattern of `Assign(x, Function())` or `Assign(x, MatchFunction())` and plumb it through `Function` too